src/sage/rings/lazy_species.py

```diff
@@ -0,0 +1,1663 @@
+           Polygon species
+        """
+        return "Polygon species"
+
+
+class OrientedSetSpecies(LazySpeciesElement, UniqueRepresentation,
+                         metaclass=InheritComparisonClasscallMetaclass):
+    def __init__(self, parent):
+        r"""
+        Initialize the species of polygons.
+
+        TESTS::
+
+            sage: L = LazySpecies(QQ, "X")
+            sage: Eo = L.OrientedSets()
+            sage: TestSuite(Eo).run(skip=['_test_category', '_test_pickling'])
+
+            sage: Eo is L.OrientedSets()
+            True
+        """
+        S = parent(AlternatingGroup, valuation=4)
+        super().__init__(parent, S._coeff_stream)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+           sage: LazySpecies(QQ, "X").OrientedSets()  # indirect doctest
+           Oriented Set species
+        """
+        return "Oriented Set species"
+
+
+class GraphSpecies(LazySpeciesElement_generating_series_mixin,
+                   LazySpeciesElement, UniqueRepresentation,
+                   metaclass=InheritComparisonClasscallMetaclass):
+    def __init__(self, parent):
+        r"""
+        Initialize the species of simple graphs.
+
+        TESTS::
+
+            sage: L = LazySpecies(QQ, "X")
+            sage: G = L.Graphs()
+            sage: TestSuite(G).run(skip=['_test_category', '_test_pickling'])
+
+            sage: G is L.Graphs()
+            True
+        """
+        P = parent._laurent_poly_ring
+        S = parent(lambda n: sum(P(G.automorphism_group()) for G in graphs(n)))
+        super().__init__(parent, S._coeff_stream)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+           sage: LazySpecies(QQ, "X").Graphs()  # indirect doctest
+           Graph species
+        """
+        return "Graph species"
+
+    def isotypes(self, labels):
+        r"""
+        Iterate over the isotypes on the given list of sizes.
+
+        EXAMPLES::
+
+            sage: L = LazySpecies(QQ, "X")
+            sage: G = L.Graphs()
+            sage: list(G.isotypes(2))
+            [Graph on 2 vertices, Graph on 2 vertices]
+        """
+        if labels in ZZ:
+            yield from (G.canonical_label().copy(immutable=True) for G in graphs(labels))
+
+    def generating_series(self):
+        r"""
+        Return the (exponential) generating series of the
+        species of simple graphs.
+
+        EXAMPLES::
+
+            sage: L = LazySpecies(QQ, "X")
+            sage: L.Graphs().generating_series().truncate(7)
+            1 + X + X^2 + 4/3*X^3 + 8/3*X^4 + 128/15*X^5 + 2048/45*X^6
+        """
+        P = self.parent()
+        L = LazyPowerSeriesRing(P.base_ring().fraction_field(),
+                                P._laurent_poly_ring._indices._indices.variable_names())
+        return L(lambda n: 2 ** binomial(n, 2) / factorial(n))
```

Comment:
```suggestion
        return L(lambda n: 2**binomial(n, 2) / factorial(n))
```
PEP8 allows this (and perhaps slightly encourages having higher precedent operators not having a space around them). Anyways, up to you.